pypeit/par/pypeitpar.py

```diff
@@ -1241,6 +1245,26 @@ def __init__(self, ex_value=None, flux_value=None, nmaskedge=None,
         dtypes['sn_min_polyscale'] = [int, float]
         descr['sn_min_polyscale'] = "For scale method set to ``auto``, this sets the minimum SNR for which polynomial scaling is attempted."
 
+        defaults['weight_method'] = 'auto'
+        dtypes['weight_method'] = str
```

Comment:
Maybe add something like the following here:

`options['weight_method'] = Coadd1DPar.valid_weight_methods()`

with the following new function added to the `Coadd1DPar()` class:

```
    @staticmethod
    def valid_weight_methods():
        """
        Return the valid options for the weighting of spectra.
        """
        return ['auto', 'constant', 'uniform', 'wave_dependent', 'relative', 'ivar']
```

I appreciate other functions have ignored this functionality before, but it's good to keep things documented like this.

Reply:
Added.